ui/components/GHImport.js

```diff
@@ -0,0 +1,155 @@
+import React from 'react'
+import { Tooltip, IconButton, TextField, Button, Grid, Typography } from '@material-ui/core';
+import { makeStyles, MuiThemeProvider } from '@material-ui/core/styles';
+import { createTheme } from '@material-ui/core/styles';
+import GitHubIcon from '@material-ui/icons/GitHub';
+import InputLabel from '@material-ui/core/InputLabel';
+import MenuItem from '@material-ui/core/MenuItem';
+import FormControl from '@material-ui/core/FormControl';
+import Select from '@material-ui/core/Select';
+import { URLValidator } from '../utils/URLValidator';
+
+const getMuiTheme = () => createTheme({
+  palette : {
+    primary : {
+      main : "#607d8b"
+    }
+  },
+  overrides : {
+    MuiGrid : {
+      input : {
+        color : '#607d8b'
+      }
+    },
+  }
+})
+
+
+const styles = makeStyles((theme) => ({
+  paper : {
+    position : 'absolute',
+    width : 600,
+    backgroundColor : theme.palette.background.paper,
+    border : '0px solid #000',
+    boxShadow : theme.shadows[5],
+    padding : theme.spacing(2, 4, 3),
+    top : '50%',
+    left : '50%',
+    transform : `translate(-50%, -50%)`,
+    borderRadius : 10,
+  },
+  grid : {
+    width : '100%'
+  },
+  GridTitle : {
+    textAlign : 'center'
+  },
+  formControl : {
+    margin : theme.spacing(1),
+    minWidth : 120,
+  },
+  ghGroups : {
+    padding : theme.spacing(0, 13, 0)
+  },
+  optionImport : {
+    textAlign : 'center'
+  }
+
+}));
+import GenericModal from "./GenericModal";
+
+
+const GHImport = ({ onSubmit }) => {
+  const classes = styles();
+  const [open, setOpen] = React.useState(false);
+  const [input, setInput] = React.useState('');
```

Comment:
We need different states here to store `owner, repo, branch, dirPath` and a boolean `recursive` to check if recursion is needed. 
Once we have that,
You should create a new state/ just a const `finalURL` using string interpolation,
set `finalURL` as       
```
`https://github.com/${owner}/${repository}/${branch}/`
```
And another `path` as 
```
`${dirPath}/${ recursive ? '**' / ''}`
```
Something like this.

And then body will be :
```
{ 
  "url": finalURL,
  “path”: path,
  “save”: false 
}
```
This should be passed as `body` parameter `onSubmit` Function from what I understand.

Reply:
okay, I'll update the pr accordingly, thanks.